lightning/src/chain/package.rs

```diff
@@ -746,16 +746,17 @@ pub struct PackageTemplate {
 	// Cache of next height at which fee-bumping and rebroadcast will be attempted. In
 	// the future, we might abstract it to an observed mempool fluctuation.
 	height_timer: u32,
-	// Confirmation height of the claimed outputs set transaction. In case of reorg reaching
-	// it, we wipe out and forget the package.
-	height_original: u32,
 }
 
 impl PackageTemplate {
 	pub(crate) fn is_malleable(&self) -> bool {
 		self.malleability == PackageMalleability::Malleable
 	}
-	pub(crate) fn timelock(&self) -> u32 {
+	/// The height at which our counterparty may be able to spend this output.
+	///
+	/// This is an important limit for aggregation as after this height our counterparty may be
+	/// able to pin transactions spending this output in the mempool.
+	pub(crate) fn counterparty_spendable_height(&self) -> u32 {
 		self.soonest_conf_deadline
```

Comment:
Perhaps we can rename the variable as well?

I was also confused why it was not an `Option`, aren't there cases where the counterparty might be able to claim it immediately? However, the `build_package` calls don't seem to discriminate between those.

Reply:
> There's always some height at which a counterparty could claim an output, assuming they have the cryptographic material with which to do so, though at least for our outputs maybe we shouldn't set one (because our counterparty can only claim if its a stale output).

True, as in `0` or `current_height` for outputs without any CSV/CLTV locks? I was wondering if it was clearer to talk about the presence of any of those locks, but it's not really necessary.

I was also confused because the `build_package` call for holder HTLC outputs seems to always set `counterparty_spendable_height` to `htlc.cltv_expiry`, regardless if it was offered/received:
https://github.com/lightningdevkit/rust-lightning/blob/cdd1298a38a71eb24b2e1fe719212f645aea5bf0/lightning/src/chain/channelmonitor.rs#L3668-L3689